File: gitdot-web/packages/api/src/endpoint/repository/get-repository-settings.ts
import { z } from "zod";

import { RepositorySettingsResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetRepositorySettingsRequest = z.object({});
export type GetRepositorySettingsRequest = z.infer<
  typeof GetRepositorySettingsRequest
>;

export const GetRepositorySettingsResponse = RepositorySettingsResource;
export type GetRepositorySettingsResponse = z.infer<
  typeof GetRepositorySettingsResponse
>;

export const GetRepositorySettings = {
  path: "/repository/{owner}/{repo}/settings",
  method: "GET",
  request: GetRepositorySettingsRequest,
  response: GetRepositorySettingsResponse,
} as const satisfies Endpoint;
export type GetRepositorySettings = typeof GetRepositorySettings;

File: gitdot-api/src/endpoint/repository/get_repository_settings.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositorySettingsResource};

pub struct GetRepositorySettings;

impl Endpoint for GetRepositorySettings {
    const PATH: &'static str = "/repository/{owner}/{repo}/settings";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRepositorySettingsRequest;
    type Response = GetRepositorySettingsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetRepositorySettingsRequest {}

pub type GetRepositorySettingsResponse = RepositorySettingsResource;
